tests/unit/test_controller.py

```diff
@@ -0,0 +1,41 @@
+import asynctest
+
+
+class TestControllerConnect(asynctest.TestCase):
+    @asynctest.patch('juju.client.connector.Connector.connect_controller')
+    async def test_controller_connect_no_args(self, mock_connect_controller):
+        from juju.controller import Controller
+        c = Controller()
+        await c.connect()
+        self.assertEqual(mock_connect_controller.call_count, 1)
```

Comment:
These two assertions can be collapsed into as single `mock_connect_controller.assert_called_once_with(None)`

Reply:
Ah, thanks, I didn't know about that.